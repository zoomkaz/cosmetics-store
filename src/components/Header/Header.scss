header {
  .dotted {
    opacity: 0.05;
    border: 1px dashed #000000;
    width: 1px;
    height: 40px;
  }
  .top {
    width: 100%;
    padding: 15px 0;
    display: flex;
    justify-content: space-between;
    align-items: center;
    .info {
      display: flex;
      gap: 20px;
      align-items: center;
      .location,
      .email {
        h3 {
          font-weight: 600;
          font-size: 14px;
          line-height: 18.2px;
        }
        p {
          font-weight: 300;
          font-size: 12px;
          line-height: 15.6px;
          color: #3f4e65;
        }
        display: flex;
        align-items: center;
        gap: 10px;
      }
    }
    ul {
      display: flex;
      align-items: center;
      gap: 30px;
      font-style: normal;
      font-weight: 400;
      font-size: 14px;
      line-height: 150%;
      color: #3f4e65;
      li {
        transition: all 0.2s linear;
        &:hover {
          cursor: pointer;
          transform: scale(1.1);
        }
        &:active {
          transform: scale(0.9);
        }
      }
    }
  }
  .bottom {
    width: 100%;
    padding: 20px 0;
    display: flex;
    align-items: center;
    justify-content: space-between;
    .logo {
      background: url("../../../img/logo.svg");
      width: 156px;
      height: 66px;
    }
    .catalog-btn,
    .price-btn {
      display: flex;
      justify-content: center;
      align-items: center;
      color: white;
      font-style: normal;
      font-weight: 700;
      font-size: 14px;
      line-height: 17px;
      padding: 20px 55px;
      gap: 10px;
      background: #ffc85e;
      border-radius: 79px;
      transition: all 0.2s linear;
      &:hover {
        transform: scale(1.05);
      }
      &:active {
        transform: scale(0.95);
      }
    }
    .catalog-btn {
      width: 192px;
      .icon {
        background: url("../../../img/catalog-btn.svg") no-repeat center center;
        width: 15px;
        height: 15px;
      }
    }
    .price-btn {
      width: 220px;
      .icon {
        background: url("../../../img/price-btn.svg") no-repeat center center;
        width: 15px;
        height: 15px;
      }
    }
    .search {
      display: flex;
      align-items: center;
      width: 263px;
      input {
        padding: 20px 10px 20px 20px;
        gap: 10px;
        background: #ededed;
        border-radius: 36px;
        transition: all 0.2s linear;
        &:focus {
          border: 1px solid #5c6370;
        }
      }
      button {
        width: 40px;
        height: 40px;
        display: flex;
        align-items: center;
        justify-content: center;
        background: #ffc85e;
        border-radius: 100%;
        margin-left: -50px;
        transition: all 0.2s linear;
        &:hover {
          transform: scale(1.05);
        }
        &:active {
          transform: scale(0.95);
        }
        .icon {
          background: url("../../../img/search.svg") no-repeat center center;
        }
      }
    }
    .info {
      display: flex;
      align-items: center;
      gap: 15px;
      .info-left {
        text-align: right;
        h3 {
          font-style: normal;
          font-weight: 600;
          font-size: 14px;
          line-height: 150%;
          color: #111111;
        }
        h4 {
          font-style: normal;
          font-weight: 300;
          font-size: 12px;
          line-height: 150%;
          color: #3f4e65;
        }
        p {
          font-style: normal;
          font-weight: 400;
          font-size: 10px;
          line-height: 150%;
          text-decoration-line: underline;
          color: #3f4e65;
          &:hover {
            cursor: pointer;
          }
        }
      }
      .info-right {
        background: url("../../../img/info-right.svg") no-repeat center center;
        width: 70px;
        height: 92px;
      }
    }
    .basket_container {
      display: flex;
      align-items: center;
      gap: 14px;
      .basket {
        position: relative;
        background: url("../../../img/basket.svg") no-repeat center center;
        width: 58px;
        height: 38px;
        transition: all 0.2s linear;
        &:hover {
          transform: scale(1.1);
        }
        &:active {
          transform: scale(1);
        }
        span {
          position: absolute;
          top: 6px;
          right: 9px;
          font-style: normal;
          font-weight: 700;
          font-size: 14px;
          line-height: 17px;
          color: #ffffff;
        }
      }
      .total_container {
        display: flex;
        flex-direction: column;
        gap: 5px;
        h3 {
          font-style: normal;
          font-weight: 300;
          font-size: 12px;
          line-height: 150%;
          color: #3f4e65;
        }
        .total {
          font-style: normal;
          font-weight: 600;
          font-size: 14px;
          line-height: 150%;
          color: #111111;
        }
      }
    }
  }
}

.icon {
  &.geo {
    background: url("../../../img/geo.svg") no-repeat center center;
  }
  &.mail {
    background: url("../../../img/mail.svg") no-repeat center center;
  }
  width: 18px;
  height: 18px;
}

.header-for-mobile {
  display: none;
}

@media (max-width: 576px) {
  header {
    display: none !important;
  }
  .header-for-mobile {
    display: block;
    position: fixed;
    background: #ffffff;
    z-index: 1000;
    width: 100%;
    .header-top {
      display: flex;
      align-items: center;
      justify-content: space-between;
      padding: 0 15px;
      .burger {
        width: 32px;
        height: 32px;
        background-color: #ffc85e;
        border-radius: 100%;
        display: flex;
        justify-content: center;
        align-items: center;
        .icon {
          background: url("../../../img/burger-hidden.svg") no-repeat center center;
          width: 8px;
          height: 8px;
        }
      }
      .logo {
        background: url("../../../img/logo.svg") no-repeat;
        background-size: contain;
        background-position: center;
        width: 100px;
        height: 40px;
      }
      .basket_container {
        display: flex;
        align-items: center;
        gap: 14px;
        .basket {
          position: relative;
          background: url("../../../img/basket.svg") no-repeat;
          background-size: contain;
          background-position: center;
          width: 25px;
          height: 25px;
          transition: all 0.2s linear;
          &:hover {
            transform: scale(1.1);
          }
          &:active {
            transform: scale(1);
          }
          span {
            position: absolute;
            top: 26.5px;
            right: 5.5px;
            font-style: normal;
            font-weight: 700;
            font-size: 7px;
            line-height: 12px;
            color: #ffffff;
          }
        }
      }
    }
    .header-bottom {
      padding: 15px;
      display: flex;
      align-items: center;
      justify-content: center;
      gap: 45px;
      .dotted {
        height: 30px;
      }
      .header-bottom-catalog {
        .link {
          display: flex;
          align-items: center;
          gap: 6px;
          .icon {
            background: url("../../../img/catalog-icon.svg") no-repeat center center;
            width: 15px;
            height: 15px;
          }
          h2 {
            font-style: normal;
            font-weight: 600;
            font-size: 12px;
            line-height: 15px;
            color: #3f4e65;
          }
        }
      }
      .header-bottom-search {
        display: flex;
        align-items: center;
        gap: 6px;
        .icon {
          background: url("../../../img/search-icon.svg") no-repeat center center;
          width: 15px;
          height: 15px;
        }
        h2 {
          font-style: normal;
          font-weight: 600;
          font-size: 12px;
          line-height: 15px;
          color: #3f4e65;
        }
      }
    }
    .show-header {
      padding: 20px 15px 32px 15px;
      box-shadow: 0px 20px 20px 0px gray;
      .dotted {
        width: 100%;
        height: 1px;
        margin-top: 25px;
      }
      .header-mobile-info {
        .location,
        .email,
        .sell {
          display: flex;
          align-items: center;
          gap: 15px;
          margin-bottom: 15px;
          .text {
            h3 {
              font-style: normal;
              font-weight: 600;
              font-size: 14px;
              line-height: 130.02%;
              color: #111111;
            }
            p {
              font-style: normal;
              font-weight: 300;
              font-size: 12px;
              line-height: 130.02%;
              color: #3f4e65;
            }
          }
        }
        .sell {
          .icon {
            background: url("../../../img/sell.svg") no-repeat center center;
            width: 15px;
            height: 15px;
          }
        }
        .order-call {
          display: flex;
          align-items: center;
          gap: 7px;
          .call {
            display: flex;
            align-items: center;
            justify-content: center;
            width: 30px;
            height: 30px;
            background: #ffc85e;
            border-radius: 100%;
            .icon {
              background: url("../../../img/call.svg") no-repeat center center;
              width: 8px;
              height: 8px;
            }
          }
          h2 {
            font-style: normal;
            font-weight: 400;
            font-size: 10px;
            line-height: 150%;
            text-decoration-line: underline;
            color: #3f4e65;
          }
        }
      }
      .nav {
        h2 {
          margin: 25px 0;
          font-style: normal;
          font-weight: 600;
          font-size: 20px;
          line-height: 150%;
          color: #3f4e65;
        }
        nav {
          margin-bottom: 22px;
          ul {
            display: flex;
            flex-direction: column;
            gap: 20px;
            li {
              font-style: normal;
              font-weight: 400;
              font-size: 14px;
              line-height: 150%;
              color: #3f4e65;
            }
          }
        }
        .price-btn {
          width: 100%;
          padding: 25px 90px;
          display: flex;
          align-items: center;
          justify-content: center;
          gap: 12px;
          font-style: normal;
          font-weight: 700;
          font-size: 14px;
          line-height: 17px;
          color: #ffffff;
          background: #ffc85e;
          border-radius: 79px;
          transition: all 0.2s linear;
          &:hover {
            transform: scale(1.1);
          }
          &:active {
            transform: scale(1);
          }
          .icon {
            background: url("../../../img/price-btn.svg") no-repeat;
            background-position: center;
            background-size: contain;
            width: 8px;
            height: 8px;
          }
        }
      }
    }
  }
}

@media (min-width: 768px) and (max-width: 1023px) {
  header {
    width: 100%;
    overflow: hidden;
    .top {
      padding: 15px;
      .info {
        gap: 10px;
        .location,
        .email {
          h3 {
            font-size: 10px;
          }
          p {
            font-size: 9px;
          }
        }
      }
      ul {
        gap: 10px;
        font-size: 10px;
      }
    }
    .bottom {
      width: 100%;
      padding: 15px;
      .catalog-btn,
      .price-btn {
        display: flex;
        justify-content: center;
        align-items: center;
        font-size: 14px;
        padding: 20px 55px;
        gap: 5px;
      }
      .catalog-btn {
        width: initial;
      }
      .price-btn {
        display: none;
        width: initial;
      }
      .search {
        display: flex;
        align-items: center;
        width: initial;
        input {
          padding: 20px 10px;
          gap: 10px;
        }
      }
      .info {
        display: flex;
        align-items: center;
        gap: 0px;
        .info-left {
          display: none;
        }
        .info-right {
          display: none;
        }
      }
      .basket_container {
        display: flex;
        align-items: center;
        gap: 5px;
      }
    }
  }
}

@media (min-width: 1024px) and (max-width: 1443px) {
  header {
    .top {
      padding: 15px;
    }
    .bottom {
      padding: 15px;
      .price-btn {
        display: none;
        width: initial;
      }
    }
  }
}
