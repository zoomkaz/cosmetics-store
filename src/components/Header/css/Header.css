header .dotted {
  opacity: 0.05;
  border: 1px dashed #000000;
  width: 1px;
  height: 40px;
}
header .top {
  width: 100%;
  padding: 15px 0;
  display: flex;
  justify-content: space-between;
  align-items: center;
}
header .top .info {
  display: flex;
  gap: 20px;
  align-items: center;
}
header .top .info .location,
header .top .info .email {
  display: flex;
  align-items: center;
  gap: 10px;
}
header .top .info .location h3,
header .top .info .email h3 {
  font-weight: 600;
  font-size: 14px;
  line-height: 18.2px;
}
header .top .info .location p,
header .top .info .email p {
  font-weight: 300;
  font-size: 12px;
  line-height: 15.6px;
  color: #3f4e65;
}
header .top ul {
  display: flex;
  align-items: center;
  gap: 30px;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 150%;
  color: #3f4e65;
}
header .top ul li {
  transition: all 0.2s linear;
}
header .top ul li:hover {
  cursor: pointer;
  transform: scale(1.1);
}
header .top ul li:active {
  transform: scale(0.9);
}
header .bottom {
  width: 100%;
  padding: 20px 0;
  display: flex;
  align-items: center;
  justify-content: space-between;
}
header .bottom .logo {
  background: url("../../../img/logo.svg");
  width: 156px;
  height: 66px;
}
header .bottom .catalog-btn,
header .bottom .price-btn {
  display: flex;
  justify-content: center;
  align-items: center;
  color: white;
  font-style: normal;
  font-weight: 700;
  font-size: 14px;
  line-height: 17px;
  padding: 20px 55px;
  gap: 10px;
  background: #ffc85e;
  border-radius: 79px;
  transition: all 0.2s linear;
}
header .bottom .catalog-btn:hover,
header .bottom .price-btn:hover {
  transform: scale(1.05);
}
header .bottom .catalog-btn:active,
header .bottom .price-btn:active {
  transform: scale(0.95);
}
header .bottom .catalog-btn {
  width: 192px;
}
header .bottom .catalog-btn .icon {
  background: url("../../../img/catalog-btn.svg") no-repeat center center;
  width: 15px;
  height: 15px;
}
header .bottom .price-btn {
  width: 220px;
}
header .bottom .price-btn .icon {
  background: url("../../../img/price-btn.svg") no-repeat center center;
  width: 15px;
  height: 15px;
}
header .bottom .search {
  display: flex;
  align-items: center;
  width: 263px;
}
header .bottom .search input {
  padding: 20px 10px 20px 20px;
  gap: 10px;
  background: #ededed;
  border-radius: 36px;
  transition: all 0.2s linear;
}
header .bottom .search input:focus {
  border: 1px solid #5c6370;
}
header .bottom .search button {
  width: 40px;
  height: 40px;
  display: flex;
  align-items: center;
  justify-content: center;
  background: #ffc85e;
  border-radius: 100%;
  margin-left: -50px;
  transition: all 0.2s linear;
}
header .bottom .search button:hover {
  transform: scale(1.05);
}
header .bottom .search button:active {
  transform: scale(0.95);
}
header .bottom .search button .icon {
  background: url("../../../img/search.svg") no-repeat center center;
}
header .bottom .info {
  display: flex;
  align-items: center;
  gap: 15px;
}
header .bottom .info .info-left {
  text-align: right;
}
header .bottom .info .info-left h3 {
  font-style: normal;
  font-weight: 600;
  font-size: 14px;
  line-height: 150%;
  color: #111111;
}
header .bottom .info .info-left h4 {
  font-style: normal;
  font-weight: 300;
  font-size: 12px;
  line-height: 150%;
  color: #3f4e65;
}
header .bottom .info .info-left p {
  font-style: normal;
  font-weight: 400;
  font-size: 10px;
  line-height: 150%;
  text-decoration-line: underline;
  color: #3f4e65;
}
header .bottom .info .info-left p:hover {
  cursor: pointer;
}
header .bottom .info .info-right {
  background: url("../../../img/info-right.svg") no-repeat center center;
  width: 70px;
  height: 92px;
}
header .bottom .basket_container {
  display: flex;
  align-items: center;
  gap: 14px;
}
header .bottom .basket_container .basket {
  position: relative;
  background: url("../../../img/basket.svg") no-repeat center center;
  width: 58px;
  height: 38px;
  transition: all 0.2s linear;
}
header .bottom .basket_container .basket:hover {
  transform: scale(1.1);
}
header .bottom .basket_container .basket:active {
  transform: scale(1);
}
header .bottom .basket_container .basket span {
  position: absolute;
  top: 6px;
  right: 9px;
  font-style: normal;
  font-weight: 700;
  font-size: 14px;
  line-height: 17px;
  color: #ffffff;
}
header .bottom .basket_container .total_container {
  display: flex;
  flex-direction: column;
  gap: 5px;
}
header .bottom .basket_container .total_container h3 {
  font-style: normal;
  font-weight: 300;
  font-size: 12px;
  line-height: 150%;
  color: #3f4e65;
}
header .bottom .basket_container .total_container .total {
  font-style: normal;
  font-weight: 600;
  font-size: 14px;
  line-height: 150%;
  color: #111111;
}

.icon {
  width: 18px;
  height: 18px;
}
.icon.geo {
  background: url("../../../img/geo.svg") no-repeat center center;
}
.icon.mail {
  background: url("../../../img/mail.svg") no-repeat center center;
}

.header-for-mobile {
  display: none;
}

@media (max-width: 576px) {
  header {
    display: none !important;
  }
  .header-for-mobile {
    display: block;
    position: fixed;
    background: #ffffff;
    z-index: 1000;
    width: 100%;
  }
  .header-for-mobile .header-top {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 0 15px;
  }
  .header-for-mobile .header-top .burger {
    width: 32px;
    height: 32px;
    background-color: #ffc85e;
    border-radius: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
  }
  .header-for-mobile .header-top .burger .icon {
    background: url("../../../img/burger-hidden.svg") no-repeat center center;
    width: 8px;
    height: 8px;
  }
  .header-for-mobile .header-top .logo {
    background: url("../../../img/logo.svg") no-repeat;
    background-size: contain;
    background-position: center;
    width: 100px;
    height: 40px;
  }
  .header-for-mobile .header-top .basket_container {
    display: flex;
    align-items: center;
    gap: 14px;
  }
  .header-for-mobile .header-top .basket_container .basket {
    position: relative;
    background: url("../../../img/basket.svg") no-repeat;
    background-size: contain;
    background-position: center;
    width: 25px;
    height: 25px;
    transition: all 0.2s linear;
  }
  .header-for-mobile .header-top .basket_container .basket:hover {
    transform: scale(1.1);
  }
  .header-for-mobile .header-top .basket_container .basket:active {
    transform: scale(1);
  }
  .header-for-mobile .header-top .basket_container .basket span {
    position: absolute;
    top: 26.5px;
    right: 5.5px;
    font-style: normal;
    font-weight: 700;
    font-size: 7px;
    line-height: 12px;
    color: #ffffff;
  }
  .header-for-mobile .header-bottom {
    padding: 15px;
    display: flex;
    align-items: center;
    justify-content: center;
    gap: 45px;
  }
  .header-for-mobile .header-bottom .dotted {
    height: 30px;
  }
  .header-for-mobile .header-bottom .header-bottom-catalog .link {
    display: flex;
    align-items: center;
    gap: 6px;
  }
  .header-for-mobile .header-bottom .header-bottom-catalog .link .icon {
    background: url("../../../img/catalog-icon.svg") no-repeat center center;
    width: 15px;
    height: 15px;
  }
  .header-for-mobile .header-bottom .header-bottom-catalog .link h2 {
    font-style: normal;
    font-weight: 600;
    font-size: 12px;
    line-height: 15px;
    color: #3f4e65;
  }
  .header-for-mobile .header-bottom .header-bottom-search {
    display: flex;
    align-items: center;
    gap: 6px;
  }
  .header-for-mobile .header-bottom .header-bottom-search .icon {
    background: url("../../../img/search-icon.svg") no-repeat center center;
    width: 15px;
    height: 15px;
  }
  .header-for-mobile .header-bottom .header-bottom-search h2 {
    font-style: normal;
    font-weight: 600;
    font-size: 12px;
    line-height: 15px;
    color: #3f4e65;
  }
  .header-for-mobile .show-header {
    padding: 20px 15px 32px 15px;
    box-shadow: 0px 20px 20px 0px gray;
  }
  .header-for-mobile .show-header .dotted {
    width: 100%;
    height: 1px;
    margin-top: 25px;
  }
  .header-for-mobile .show-header .header-mobile-info .location,
  .header-for-mobile .show-header .header-mobile-info .email,
  .header-for-mobile .show-header .header-mobile-info .sell {
    display: flex;
    align-items: center;
    gap: 15px;
    margin-bottom: 15px;
  }
  .header-for-mobile .show-header .header-mobile-info .location .text h3,
  .header-for-mobile .show-header .header-mobile-info .email .text h3,
  .header-for-mobile .show-header .header-mobile-info .sell .text h3 {
    font-style: normal;
    font-weight: 600;
    font-size: 14px;
    line-height: 130.02%;
    color: #111111;
  }
  .header-for-mobile .show-header .header-mobile-info .location .text p,
  .header-for-mobile .show-header .header-mobile-info .email .text p,
  .header-for-mobile .show-header .header-mobile-info .sell .text p {
    font-style: normal;
    font-weight: 300;
    font-size: 12px;
    line-height: 130.02%;
    color: #3f4e65;
  }
  .header-for-mobile .show-header .header-mobile-info .sell .icon {
    background: url("../../../img/sell.svg") no-repeat center center;
    width: 15px;
    height: 15px;
  }
  .header-for-mobile .show-header .header-mobile-info .order-call {
    display: flex;
    align-items: center;
    gap: 7px;
  }
  .header-for-mobile .show-header .header-mobile-info .order-call .call {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 30px;
    height: 30px;
    background: #ffc85e;
    border-radius: 100%;
  }
  .header-for-mobile .show-header .header-mobile-info .order-call .call .icon {
    background: url("../../../img/call.svg") no-repeat center center;
    width: 8px;
    height: 8px;
  }
  .header-for-mobile .show-header .header-mobile-info .order-call h2 {
    font-style: normal;
    font-weight: 400;
    font-size: 10px;
    line-height: 150%;
    text-decoration-line: underline;
    color: #3f4e65;
  }
  .header-for-mobile .show-header .nav h2 {
    margin: 25px 0;
    font-style: normal;
    font-weight: 600;
    font-size: 20px;
    line-height: 150%;
    color: #3f4e65;
  }
  .header-for-mobile .show-header .nav nav {
    margin-bottom: 22px;
  }
  .header-for-mobile .show-header .nav nav ul {
    display: flex;
    flex-direction: column;
    gap: 20px;
  }
  .header-for-mobile .show-header .nav nav ul li {
    font-style: normal;
    font-weight: 400;
    font-size: 14px;
    line-height: 150%;
    color: #3f4e65;
  }
  .header-for-mobile .show-header .nav .price-btn {
    width: 100%;
    padding: 25px 90px;
    display: flex;
    align-items: center;
    justify-content: center;
    gap: 12px;
    font-style: normal;
    font-weight: 700;
    font-size: 14px;
    line-height: 17px;
    color: #ffffff;
    background: #ffc85e;
    border-radius: 79px;
    transition: all 0.2s linear;
  }
  .header-for-mobile .show-header .nav .price-btn:hover {
    transform: scale(1.1);
  }
  .header-for-mobile .show-header .nav .price-btn:active {
    transform: scale(1);
  }
  .header-for-mobile .show-header .nav .price-btn .icon {
    background: url("../../../img/price-btn.svg") no-repeat;
    background-position: center;
    background-size: contain;
    width: 8px;
    height: 8px;
  }
}
@media (min-width: 768px) and (max-width: 1023px) {
  header {
    width: 100%;
    overflow: hidden;
  }
  header .top {
    padding: 15px;
  }
  header .top .info {
    gap: 10px;
  }
  header .top .info .location h3,
  header .top .info .email h3 {
    font-size: 10px;
  }
  header .top .info .location p,
  header .top .info .email p {
    font-size: 9px;
  }
  header .top ul {
    gap: 10px;
    font-size: 10px;
  }
  header .bottom {
    width: 100%;
    padding: 15px;
  }
  header .bottom .catalog-btn,
  header .bottom .price-btn {
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 14px;
    padding: 20px 55px;
    gap: 5px;
  }
  header .bottom .catalog-btn {
    width: initial;
  }
  header .bottom .price-btn {
    display: none;
    width: initial;
  }
  header .bottom .search {
    display: flex;
    align-items: center;
    width: initial;
  }
  header .bottom .search input {
    padding: 20px 10px;
    gap: 10px;
  }
  header .bottom .info {
    display: flex;
    align-items: center;
    gap: 0px;
  }
  header .bottom .info .info-left {
    display: none;
  }
  header .bottom .info .info-right {
    display: none;
  }
  header .bottom .basket_container {
    display: flex;
    align-items: center;
    gap: 5px;
  }
}
@media (min-width: 1024px) and (max-width: 1443px) {
  header .top {
    padding: 15px;
  }
  header .bottom {
    padding: 15px;
  }
  header .bottom .price-btn {
    display: none;
    width: initial;
  }
}