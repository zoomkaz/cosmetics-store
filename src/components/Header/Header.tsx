import React, { useEffect, useState } from 'react'
import { Link } from 'react-router-dom'
import './css/Header.css'
import burgerS from '../../img/burger-showed.svg'
import burgerH from '../../img/burger-hidden.svg'

export const Header = () => {
  const [total, setTotal] = useState(0)
  const [orders, setOrders] = useState(0)
  const [showHeader, setShowHeader] = useState(false)

  useEffect(() => {
    let total = JSON.parse(localStorage.getItem('info')!)?.total
    setTotal(total ? total.toFixed(2) : 0)
    let orders = JSON.parse(localStorage.getItem('info')!)?.orders
    setOrders(orders ? orders : 0)

    let burger = document.querySelector('.burger .icon');
    if (showHeader) {
      (burger as HTMLElement).style.background = `url(${burgerS}) no-repeat center center`;
    } else if (!showHeader) {
      (burger as HTMLElement).style.background = `url(${burgerH}) no-repeat center center`;
    }
  }, [total, orders, showHeader])


  const showHeaderMobile = (e: any) => {
    showHeader ? setShowHeader(false) : setShowHeader(true);
  }


  return (
    <>
      <header>
        <div className="top content">
          <div className="info">
            <div className="location">
              <div className="icon geo"></div>
              <div className="text">
                <h3>г. Кокчетав, ул. Ж. Ташенова 129Б</h3>
                <p>(Рынок Восточный)</p>
              </div>
            </div>
            <div className="dotted"></div>
            <div className="email">
              <div className="icon mail"></div>
              <div className="text">
                <h3>opt.sultan@mail.ru</h3>
                <p>На связи в любое время</p>
              </div>
            </div>
          </div>
          <nav>
            <ul>
              <li>О компании</li>
              <div className="dotted"></div>
              <li>Доставка и оплата</li>
              <div className="dotted"></div>
              <li>Возврат</li>
              <div className="dotted"></div>
              <li>Контакты</li>
            </ul>
          </nav>
        </div>
        <div className="line"></div>
        <div className="bottom content">
          <Link to='/' onClick={() => localStorage.clear()}><div className="logo"></div></Link>
          <Link to='/'><button className='catalog-btn'>
            Каталог
            <div className="icon"></div>
          </button></Link>
          <div className="search">
            <input type="text" placeholder='Поиск...' />
            <button><div className="icon"></div></button>
          </div>
          <div className="info">
            <div className="info-left">
              <h3>+7 (777) 490-00-91</h3>
              <h4>время работы: 9:00-20:00</h4>
              <p>Заказать звонок</p>
            </div>
            <div className="info-right"></div>
          </div>
          <button className='price-btn'>
            Прайс-лист
            <div className="icon"></div>
          </button>
          <div className="basket_container">
            <Link to='/basket'><div className="basket"><span>{orders}</span></div></Link>
            <div className="total_container">
              <h3>Корзина</h3>
              <h2 className='total'><span>{total}</span> ₸</h2>
            </div>
          </div>
        </div>
        <div className="line"></div>
      </header>
      <div className="header-for-mobile">
        <div className="header-top">
          <div className="burger" onClick={showHeaderMobile}>
            <div className="icon"></div>
          </div>
          <a href='/'><div className="logo"></div></a>
          <div className="basket_container">
            <Link to='/basket'><div className="basket"><span>{orders}</span></div></Link>
          </div>
        </div>
        <div className="line"></div>
        <div className="header-bottom">
          <div className="header-bottom-catalog">
            <Link className='link' to='/'>
              <div className="icon"></div>
              <h2>Каталог</h2>
            </Link>
          </div>
          <div className="dotted"></div>
          <div className="header-bottom-search">
            <div className="icon"></div>
            <h2>Поиск</h2>
          </div>
        </div>
        <div className="line"></div>
        {showHeader ?
          <div className="show-header">
            <div className="header-mobile-info">
              <div className="location">
                <div className="icon geo"></div>
                <div className="text">
                  <h3>г. Кокчетав, ул. Ж. Ташенова 129Б</h3>
                  <p>(Рынок Восточный)</p>
                </div>
              </div>
              <div className="email">
                <div className="icon mail"></div>
                <div className="text">
                  <h3>opt.sultan@mail.ru</h3>
                  <p>На связи в любое время</p>
                </div>
              </div>
              <div className="sell">
                <div className="icon sell"></div>
                <div className="text">
                  <h3>Отдел продаж</h3>
                  <p>+7 (777) 490-00-91</p>
                  <p>время работы: 9:00-20:00</p>
                </div>
              </div>
              <div className="order-call">
                <div className="call">
                  <div className="icon"></div>
                </div>
                <h2>Заказать звонок</h2>
              </div>
            </div>
            <div className="dotted"></div>
            <div className="nav">
              <h2>Меню сайта:</h2>
              <nav>
                <ul>
                  <li>О компании</li>
                  <li>Доставка и оплата</li>
                  <li>Возврат</li>
                  <li>Контакты</li>
                </ul>
              </nav>
              <button className='price-btn'>
                Прайс-лист
                <div className="icon"></div>
              </button>
            </div>
          </div>
          :
          <></>}
      </div>
    </>
  )
}
