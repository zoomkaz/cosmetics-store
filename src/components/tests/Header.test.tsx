import React from 'react'
import '@testing-library/jest-dom'
import { HashRouter } from 'react-router-dom';
import { render, screen } from "@testing-library/react";
import { Header } from "../Header/Header";

describe('Test Header', () => {
  test('renders header elements', () => {
    render(<HashRouter><Header /></HashRouter>);
    const btnElems = screen.getAllByRole('button');
    const inputElem = screen.getByPlaceholderText(/поиск.../i);
    const topText = screen.getAllByText(/о компании/i);
    expect(btnElems[0]).toBeInTheDocument()
    expect(inputElem).toBeInTheDocument()
    expect(topText[0]).toMatchSnapshot()
  })

  test('renders header elements', async () => {
    render(<HashRouter><Header /></HashRouter>);
    const someElem = await screen.findByText(/на связи в любое время/i)
    expect(someElem).toBeInTheDocument()
  })

  test('check btn color', () => {
    render(<HashRouter><Header /></HashRouter>);
    const btnElem = screen.getAllByRole('button')
    expect(btnElem[0]).toHaveStyle({ background: `rgb(255, 200, 94);` })
  })
})