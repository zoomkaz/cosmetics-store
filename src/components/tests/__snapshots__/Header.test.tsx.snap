// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header renders header elements 1`] = `
<li>
  О компании
</li>
`;
